fcrepo-http-api/src/main/java/org/fcrepo/http/api/ContentExposingResource.java

```diff
@@ -602,10 +585,16 @@ protected void replaceResourceWithStream(final FedoraResource resource,
     }
 
     protected void patchResourcewithSparql(final FedoraResource resource,
-                                           final String requestBody,
-                                           final RdfStream resourceTriples)
-            throws MalformedRdfException, AccessDeniedException {
-        resource.updateProperties(translator(), requestBody, resourceTriples);
+            final String requestBody,
+            final RdfStream resourceTriples)
+                    throws MalformedRdfException, AccessDeniedException {
+        if (resource instanceof NonRdfSourceDescription) {
+            // update the description instead
```

Comment:
We need to take this to a better forum than comments on an outdated diff.
Otherwise, your understanding is not supported by any of the conversation so far. For example, neither of the tickets about subjects of triples in `fcr:metadata` (1540 or 1635) demand that the subjects of the triples in question be `fcr:metadata`, as @escowles describes. They both demand that the subjects in question be the _NonRDFSource, instead_.
In other words, we don't have a coherent policy here that we can implement and we need to discuss this more publicly to get one.